### DISCUSSION TITLE
Plans for this package

### DISCUSSION THREAD

**Discussion by Martin-Jung** [iiasa/rN2000#4] [Ideas]
Some brainstorming on functionalities for this package. Generally It should be a wrapper for functions that we need for various projects and that interact with the Natura 2000 system. 

**Necessary functionalities:**
- [x] Function to download N2000 database in a  specified folder
- [x] Load and format N2000 database
- [ ] Extract occurrence and/or abundance information for individual species, habitats or management types
- [ ] Export individual species and habitat occurrence information as shapefile or raster
- [x] Ensure documentation is existing for each function

**Optional:**
- [ ] Prettify the documentation
- [x] Enable an interactive leaflet visualization of N2000 sites and generated subsets
- [ ] Prepare and release to CRAN

Anything else?

**Comment by matthewlewis896** [comment]
Have presumptively named a function `n2000_extract()` in my branch which takes a spatial input and extracts which N2K sites intersect/are contained within. Can change that name though as extract may make more sense elsewhere.

**Reply by Martin-Jung** [reply]
Good job. Maybe rename to `n2000_intersect` ?

**Reply by Martin-Jung** [reply]
By the way, I have set you also as Maintainer of the Repo, so feel free to merge your changes with the master branch if you think they have matured enough

**Reply by Martin-Jung** [reply]
@matthewlewis896 Remember to tick boxes (or add new ones) on the necessary and optional functions!

**Comment by matthewlewis896** [comment]
- [ ] Fix taxonomy
- [ ] Add presence absence as well as presence only to plotting
- [ ] Add abundance to plotting - options min, max, mean, data driven estimate (habitat availability, demography data)
- [ ] Pressures - not essential but create function for later use
- [ ] Habitat type occurrence - not essential but create function for later use
